### PR TITLE
feat(controls): add pagination controls

### PR DESCRIPTION
**Pull request by eddier**
# feature description

Adding pagination controls to the the set of available components

# content

Includes the PaginationControl element as well as the necessary CSS.  Also includes testable storybook component.  No new dependencies.

# usage

Handle the navigateToPage call:

`navigateToPage(page) {
    this.setState({ currentNotes: this.getNotes(page) });
  }`

In this example you can fetch the appropriate notes from the API or if they are all loaded in a parent array just set them...

`  getNotes(pageNum) {
   return this.state.allNotes.slice(
pageNum * this.state.perPage - this.state.perPage, pageNum * this.state.perPage);
  }
`

### CODE REVIEW

**Comment by cdaringe** [comment]
**[Storybook Hub](https://hub.getstorybook.io/)** is building storybooks for every commit in this PR.

**Latest Storybook: https://38dfe732-fe93-4b01-8e82-343b109a5f83.sbook.io/**
<sub>( Visit [here](https://hub.getstorybook.io/app/github.com/Tripwire/octagon/branch/pagination/) for more info )</sub>

> _This comment is automatically added by Storybook Hub_

File: src/components/PaginationControl.jsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import React from 'react'
+import Flexbox from 'flexbox-react'
+import '../styles/components/pagination-control.css'
+class PaginationControl extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      isPrevPageValid: false,
+      isNextPageValid: true,
```
Comment (cdaringe) [inline src/components/PaginationControl.jsx]: i think this is fine, just brainstorming.  both of these ^^ are derived state.  hence, something akin to `get isPrevPageValid () { this.state.currentPage > 1 }` and the like could _feasibly_ 🔥 down some code!  no action requested.  just talkin :)

File: stories/Welcome.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+          />
+        </p>
+        <p>
+          Usually we create stories with smaller UI components in the app.<br />
+          Have a look at the <a style={styles.link} href="https://getstorybook.io/docs/basics/writing-stories" target="_blank">Writing Stories</a> section in our documentation.
+        </p>
+      </div>
+    );
+  }
+}
```
Comment (cdaringe) [inline stories/Welcome.js]: this should be removed, i think. we already have a Welcome component

**Comment by cdaringe** [comment]
hey @eddier, looks good.

- i had a few remarks.  will you 👀 at those?
- can you see the build error messages?  will want to 💚 that little guy pre merge!
- on the Pagination component,
  - i was clicking the page buttons and the text kept selecting/highlighting (because of my aggressivf clicking!).  [maybe we can drop these on?](http://caniuse.com/#feat=user-select-none)?  what do you think?
  - were things like "jump to end/start" or "select items per page" requirements?  just askin!

thanks!  this is cool stuff!  @antoinejuhel, 👀 storybook

**Comment by cdaringe** [comment]
oh, and, https://github.com/semantic-release/semantic-release#default-commit-message-format .  dont worry about rebasing.  were still learning it too!

**Comment by eddier** [comment]
Can you give me account on your Codeship instance so I can see the build issues?

I'll check out the triple click issue that is causing the highlighting, good catch.  The "jump to end" state was not part of the initial scope but it makes sense to include those as optional controls.  I'll bring it up to the design team.  I assume you guys will handle the required API support necessary for those components?

**Comment by cdaringe** [comment]
cool.  hmm.  the project should be open. i just double checked.  do you have an acct?

**Comment by cdaringe** [comment]
& yep, we'll wire in the API calls. thx!

**Comment by eddier** [comment]
eraffaele at g mail - it currently throws an error about access being denied to CS.

**Comment by jhegg** [comment]
@eddier `yarn run lint` should show the errors, until you get access to CS. I'm able to repro locally.

```
$ yarn run lint
yarn run v0.20.3
$ standard 'src/**/*.js' '.storybook/**/*.js' 'scripts/**/*.js' 'test/**/*.js' 'stories/**/*.js' 
standard: Use JavaScript Standard Style (http://standardjs.com)
standard: Run `standard --fix` to automatically fix some problems.
  /data/src/octagon/stories/controls.stories.js:36:30: Trailing spaces not allowed.
  /data/src/octagon/stories/controls.stories.js:39:18: Missing space before function parentheses.
  /data/src/octagon/stories/controls.stories.js:39:20: Missing space before opening brace.
  /data/src/octagon/stories/controls.stories.js:40:59: Extra semicolon.
  /data/src/octagon/stories/controls.stories.js:45:17: Expected consistent spacing
  /data/src/octagon/stories/controls.stories.js:144:25: Trailing spaces not allowed.
  /data/src/octagon/stories/controls.stories.js:145:49: Trailing spaces not allowed.
  /data/src/octagon/stories/controls.stories.js:146:20: Trailing spaces not allowed.
  /data/src/octagon/stories/controls.stories.js:147:34: Trailing spaces not allowed.
  /data/src/octagon/stories/controls.stories.js:148:33: Trailing spaces not allowed.
  /data/src/octagon/stories/controls.stories.js:153:1: Trailing spaces not allowed.
error Command failed with exit code 1.
```

**Comment by eddier** [comment]
Thanks Josh! Didn't realize the .stories.js files were in scope of the linter.

**Comment by cdaringe** [comment]
Access granted.  Sorry for the delay

**Comment by jhegg** [comment]
I noticed that the "5 of 15" page number text has the text selection cursor when mousing over; should that just have the normal pointing mouse cursor instead? (I'm really not sure)

**Comment by eddier** [comment]
That makes sense to me, updated to reflect that change.

**Comment by antoinejuhel** [comment]
Because of the sorting functions we have for consoles, I don't see a need for jump to end/beginning in this use case. That said if these are the pagination controls we want to use throughout our products, then yes, a jump to end/beginning would be required (a good example of this would be for reporting).

**Pull request merged**